pkg/kubectl/cmd/util/config_flags.go

```diff
@@ -0,0 +1,215 @@
+	}
+	if f.CacheDir != nil {
+		cmd.Flags().StringVar(f.CacheDir, FlagHTTPCacheDir, *f.CacheDir, "Default HTTP cache directory")
+	}
+
+	// add config options
+	if f.CertFile != nil {
+		cmd.Flags().StringVar(f.CertFile, FlagCertFile, *f.CertFile, "Path to a client certificate file for TLS")
+	}
+	if f.KeyFile != nil {
+		cmd.Flags().StringVar(f.KeyFile, FlagKeyFile, *f.KeyFile, "Path to a client key file for TLS")
+	}
+	if f.BearerToken != nil {
+		cmd.Flags().StringVar(f.BearerToken, FlagBearerToken, *f.BearerToken, "Bearer token for authentication to the API server")
+	}
+	if f.Impersonate != nil {
+		cmd.Flags().StringVar(f.Impersonate, FlagImpersonate, *f.ImpersonateGroup, "Username to impersonate for the operation")
+	}
+	if f.ImpersonateGroup != nil {
+		cmd.Flags().StringVar(f.ImpersonateGroup, FlagImpersonateGroup, *f.ImpersonateGroup, "Group to impersonate for the operation, this flag can be repeated to specify multiple groups.")
+	}
+	if f.Username != nil {
+		cmd.Flags().StringVar(f.Username, FlagUsername, *f.Username, "Username for basic authentication to the API server")
+	}
+	if f.Password != nil {
+		cmd.Flags().StringVar(f.Password, FlagPassword, *f.Password, "Password for basic authentication to the API server")
+	}
+	if f.ClusterName != nil {
+		cmd.Flags().StringVar(f.ClusterName, FlagClusterName, *f.ClusterName, "The name of the kubeconfig cluster to use")
+	}
+	if f.AuthInfoName != nil {
+		cmd.Flags().StringVar(f.AuthInfoName, FlagAuthInfoName, *f.AuthInfoName, "The name of the kubeconfig user to use")
+	}
+	if f.Namespace != nil {
+		cmd.Flags().StringVarP(f.Namespace, FlagNamespace, "n", *f.Namespace, "If present, the namespace scope for this CLI request")
+	}
+	if f.Context != nil {
+		cmd.Flags().StringVar(f.Context, FlagContext, *f.Context, "The name of the kubeconfig context to use")
+	}
+
+	if f.APIServer != nil {
+		cmd.Flags().StringVar(f.APIServer, FlagAPIServer, *f.APIServer, "The address and port of the Kubernetes API server")
+	}
+	if f.Insecure != nil {
+		cmd.Flags().StringVar(f.Insecure, FlagInsecure, *f.Insecure, "If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure")
+	}
+	if f.CertFile != nil {
+		cmd.Flags().StringVar(f.CertFile, FlagCertFile, *f.CertFile, "Path to a client certificate file for TLS")
+	}
+	if f.CAFile != nil {
+		cmd.Flags().StringVar(f.CAFile, FlagCAFile, *f.CAFile, "Path to a cert file for the certificate authority")
+	}
+	if f.Timeout != nil {
+		cmd.Flags().StringVar(f.Timeout, FlagTimeout, *f.Timeout, "The length of time to wait before giving up on a single server request. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests.")
+	}
+
+	if f.EmbedCerts != nil {
+		// TODO: there is no flag-info present for this flag in clientcmd/overrides.go
+	}
+}
+
+func NewConfigFlags(disableUsernamePassword bool) *ConfigFlags {
```

Comment:
Ah, ok.  Two flavors here. `NewWithDeprecatedPasswordFlag` that has non-nil, username and password.  and `NewConfigFlags()` which has nil username and password